### DISCUSSION TITLE
p-table with virtualScrolling and lazyLoading

### DISCUSSION THREAD

**Discussion by DanielGarashi** [primefaces/community#1015] [PrimeNG] [answered]
i have the following interfaces:

export interface A{
  key: Key;
  listValues: Value[];
}

export interface Key{
  id: string;
  user: string;
  objectName: string;
  creationDate: Date;
}

export interface Value{
  id: number;
  old: string;
  new: string;
}

I'm showing an A list in p-table:
<p-table [value]="A"
           dataKey="id"
           [rows]="10"
           [scrollable]="true"
           [virtualScroll]="true"
           [lazy]="true"
           (onLazyLoad)="load($event)">

on the ngOnInit method, I init the A array with data from the server.
i want the virtual scrolling will relate to the listValues amount in the A object and not the single A object, so when I trigger the load for the lazyLoading it will display the records in the table by the listValues correctly

notes:
* there are side effects such as: jumping to the middle of the table when trying to reach to end of the table
* I don't want to change the structure of the objects

**Comment by cetincakiroglu** [comment]
Hi,

Could you please create a stackblitz reproducer so we can identify the issue?

**Comment by DanielGarashi** [comment]
**[stackblitz](https://stackblitz.com/edit/primeng-tablevirtualscroll-demo-caxywp)**


Thanks in advance

**Reply by cetincakiroglu** [reply, marked as answer]
Hi,

VirtualScroll does not support rowspan. To use virtual scroll, please remove the rowSpan because the VirtualScroll feature requires all rows to be of the same size.

**Reply by DanielGarashi** [reply]
I removed the rowSpan and it still doesn't work properly. unable to reach the end of the table

update [stackblitz](https://stackblitz.com/edit/primeng-tablevirtualscroll-demo-caxywp)

**Reply by richard-kirk-np** [reply]
I am having the same issue as shown in your stackblitz. 

I haven't been able to find a solution so far but can see an extra 'scroll' event that is causing the issue. The issue occurs for me in prime-eng-scroller.mjs after onScrollChange(...). If the isRangeChanged is set to true, then the new values are set. The next thing that seems to happen is another scroll event that is moving the scroll bar back up. I can't find what causes this extra scroll event though so I am also stuck. It seems like the virtual scroll position gets recalculated using an incorrect number of total items meaning the scroll bar moves when it shouldn't.

**Reply by richard-kirk-np** [reply]
I have done a bit more digging into the issue I am seeing and for now I am going to use the `appendOnly = true` workaround as mentioned in this item: https://github.com/primefaces/primeng/issues/13295. This is not ideal as this will store a large number of items in the table that are no longer being displayed but at least the scrolling will work as needed.
